client/src/main/java/org/spine3/base/MapStringifier.java

```diff
@@ -0,0 +1,243 @@
+        return result;
+    }
+
+    @Override
+    protected Map<K, V> fromString(String s) {
+        final String escapedString = escaper.escape(s);
+        final String[] buckets = escapedString.split(bucketPattern);
+        final Map<K, V> resultMap = newHashMap();
+
+        for (String bucket : buckets) {
+            final Map.Entry<K, V> convertedBucket = convert(bucket);
+            resultMap.put(convertedBucket.getKey(), convertedBucket.getValue());
+        }
+        return resultMap;
+    }
+
+    private Map.Entry<K, V> convert(String bucketToConvert) {
+        final String[] keyValue = bucketToConvert.split(keyValuePattern);
+        checkKeyValue(keyValue);
+
+        final String key = unquote(keyValue[0]);
+        final String value = unquote(keyValue[1]);
+
+        try {
+            final K convertedKey = convert(keyClass, key);
+            final V convertedValue = convert(valueClass, value);
+            final Map.Entry<K, V> convertedBucket =
+                    new AbstractMap.SimpleEntry<>(convertedKey, convertedValue);
+            return convertedBucket;
+        } catch (Throwable e) {
+            throw newIllegalArgumentException("The exception is occurred during the conversion", e);
+        }
+    }
+
+    @SuppressWarnings("unchecked") // It is safe because the type is checked before the cast.
+    private static <I> I convert(Class<I> elementClass, String elementToConvert) {
+        if (isString(elementClass)) {
+            return (I) elementToConvert;
+        }
+
+        final I convertedValue = Stringifiers.fromString(elementToConvert, elementClass);
+        return convertedValue;
+
+    }
+
+    private static void checkKeyValue(String[] keyValue) {
+        if (keyValue.length != 2 || !isQuotedKeyValue(keyValue)) {
+            final String exMessage =
+                    "Illegal key-value format. The key-value should be quoted " +
+                    "and separated with the `" + KEY_VALUE_DELIMITER + "` character.";
+            throw newIllegalArgumentException(exMessage);
+        }
+    }
+
+    private static boolean isString(Class<?> aClass) {
+        return String.class.equals(aClass);
+    }
+
+    private static String unquote(String value) {
+        final String unquotedValue = Pattern.compile("\\\\")
+                                            .matcher(value.substring(2, value.length() - 2))
+                                            .replaceAll("");
+        return unquotedValue;
+    }
+
+    private static boolean isQuotedKeyValue(String[] keyValue) {
+        final String key = keyValue[0];
+        final String value = keyValue[1];
+
+        final int keyLength = key.length();
+        final int valueLength = value.length();
+
+        if (keyLength < 2 || valueLength < 2) {
+            return false;
+        }
+
+        final boolean result = isQuote(key.charAt(1)) && isQuote(key.charAt(keyLength - 1)) &&
```

Comment:
I would add a utility method called `isQuotedString()`. Please check for such a thing already available.